folly/test/ReplaceableTest.cpp

```diff
@@ -70,6 +70,14 @@ struct OddA {
 struct Indestructible {
   ~Indestructible() = delete;
 };
+
+struct HasInt {
+  HasInt(int v) : value{v} {}
+  int value{};
+};
+void swap(HasInt& lhs, HasInt& rhs) noexcept {
+  std::swap(lhs.value, rhs.value);
+}
```

Comment:
Reviewer suggests

> can we add 1) a namespace here as well just so we explicitly test the ADL case and 2) a swap that is not noexcept to test the failure case?

Reply:
Sure, I can write those tests tonight. I agree they would be good to have.

Reply:
Wrote both of those tests and updated the commit message. We now test both positive/negative cases where `Replaceable<T>` is noexcept in addition to a separate ADL-swap test to verify the user-provided `swap` function is called.

Reply:
Nice!